File: Eunleelee/2024-02/2월 17일/체육복.py
def solution(n, lost, reserve):
    answer = 0
    lost_s = list(set(lost) - set(reserve))
    reserve_s = list(set(reserve) - set(lost))
    num = len(lost_s)  
    for i in range(len(lost_s)):
        if lost_s[i] - 1 in reserve_s:
            reserve_s.remove(lost_s[i] - 1)
            num -= 1
        elif lost_s[i] + 1 in reserve_s:
            reserve_s.remove(lost_s[i] + 1)
            num -= 1
    answer = n - num
    return answer

File: jja6312/2월17일/체육복.java
class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = 0;

        int[] allStudent = new int[n];

        for (int i = 0; i < n; i++) {
            allStudent[i] = 1;
        }

        for (int x : lost) {
            allStudent[x - 1]--;
        }

        for (int plus : reserve) {
            allStudent[plus - 1]++;
        }

        for (int i = 0; i < n; i++) {
            if (allStudent[i] == 0) {// 체육복이없다면
                if (i != 0 && allStudent[i - 1] == 2) {// 왼쪽에서 빌리거나
                    allStudent[i - 1]--;
                    allStudent[i]++;
                } else if (i != n - 1 && allStudent[i + 1] == 2) {// 오른쪽에서 빌리자.
                    allStudent[i + 1]--;
                    allStudent[i]++;
                }
            }
        }

        for (int x : allStudent) {
            if (x >= 1) {
                answer++;
            }
        }

        return answer;
    }
}